StripeFinancialConnections/StripeFinancialConnections/Source/Native/PartnerAuth/PartnerAuthDataSource.swift

```diff
@@ -63,6 +64,34 @@ final class PartnerAuthDataSourceImplementation: PartnerAuthDataSource {
         }
     }
 
+    func clearReturnURL(authSession: FinancialConnectionsAuthSession, authURL: String) -> Future<FinancialConnectionsAuthSession> {
+        let promise = Promise<FinancialConnectionsAuthSession>()
+
+        apiClient
+            .generateSessionManifest(clientSecret: clientSecret, returnURL: nil)
+            .observe { [weak self] result in
+                guard let self = self else { return }
+                switch result {
+                case .success:
+                    let copiedSession = FinancialConnectionsAuthSession(id: authSession.id,
+                                                                        flow: authSession.flow,
+                                                                        institutionSkipAccountSelection: authSession.institutionSkipAccountSelection,
+                                                                        nextPane: authSession.nextPane,
+                                                                        showPartnerDisclosure: authSession.showPartnerDisclosure,
+                                                                        skipAccountSelection: authSession.skipAccountSelection,
+                                                                        url: authURL,
+                                                                        isOauth: authSession.isOauth,
+                                                                        display: authSession.display)
+                    self.pendingAuthSession = copiedSession
```

Comment:
@kgaidis-stripe is this needed? I think so?

Reply:
seems reasonable